src/libsync/vfs/cfapi/cfapiwrapper.cpp

```diff
@@ -50,11 +51,24 @@ void cfApiSendTransferInfo(const CF_CONNECTION_KEY &connectionKey, const CF_TRAN
     opParams.TransferData.CompletionStatus = status;
     opParams.TransferData.Buffer = buffer;
     opParams.TransferData.Offset.QuadPart = offset;
-    opParams.TransferData.Length.QuadPart = length;
+    opParams.TransferData.Length.QuadPart = currentBlockLength;
 
-    const qint64 result = CfExecute(&opInfo, &opParams);
-    if (result != S_OK) {
-        qCCritical(lcCfApiWrapper) << "Couldn't send transfer info" << QString::number(transferKey.QuadPart, 16) << ":" << result << QString::fromWCharArray(_com_error(result).ErrorMessage());
+    const qint64 cfExecuteresult = CfExecute(&opInfo, &opParams);
+    if (cfExecuteresult != S_OK) {
+        qCCritical(lcCfApiWrapper) << "Couldn't send transfer info" << QString::number(transferKey.QuadPart, 16) << ":" << cfExecuteresult << QString::fromWCharArray(_com_error(cfExecuteresult).ErrorMessage());
+    }
+
+    // refresh Windows Copy Dialog progress
```

Comment:
This needs to be conditional, if the `status` passed in parameter is not `STATUS_SUCCESS` there's no progress to report. This would happen on the server timing out during a download or such.